kedro/io/core.py

```diff
@@ -587,21 +588,28 @@ def parse_dataset_definition(
         class_paths = (prefix + dataset_type for prefix in _DEFAULT_PACKAGES)
 
         for class_path in class_paths:
-            tmp = _load_obj(class_path)
+            tmp, warning = _load_obj(
+                class_path
+            )  # Load dataset class, capture the warning
+
             if tmp is not None:
                 class_obj = tmp
                 break
-        else:
+
+        if class_obj is None:  # If no valid class was found, raise an error
+            error_msg = (
+                warning
+                if warning
+                else f"Class '{dataset_type}' not found, is this a typo?"
```

Comment:
I'm not sure this error will now ever be reached. I played around with it and if I have a class called `MyCSVDataset` but try to import `YourCSVDataset` in the catalog I get:

```
DatasetError: An exception occurred when parsing config for dataset 'companies':
module 'wizard.my_dataset' has no attribute 'YourCSVDataset'
```

Reply:
I think it's still better in terms of overall experience compared to before, but maybe it's still possible to catch the class error properly?